Upgrade server uyuni #11389 keichwa (#379)
uyuni-project/uyuni-docs: modules/upgrade/pages/server-major-upgrade-uyuni.adoc

[server-major-upgrade]
= Server - Major Upgrade

> If {productname} core compontents are upgraded to new major versions, {productname} Server upgrades are considered as major.

Reviewer: ```suggestion
When {productname} core components are upgraded to new major versions, you need to perform a major upgrade on the {productname} Server.
```

> This is the case, if a version upgrade of PostgreSQL, Salt, or openSUSE Leap is needed.

Reviewer: ```suggestion
This is the case if a version upgrade of PostgreSQL, Salt, or openSUSE Leap is needed.
```

openSUSE Leap is the underlying base operating system (OS).

> For background information about upgrading openSUSE Leap, see the Startup Guide at https://doc.opensuse.org.

Reviewer: ```suggestion
For more information about upgrading openSUSE Leap, see the Startup Guide at https://doc.opensuse.org.
```

Reviewer: I feel like this stuff about openSUSE Leap is a bit distracting. How relevant is it really?

[IMPORTANT]
====

> For more information, see the release notes at https://www.uyuni-project.org/pages/stable-version.html.
> Reading the release notes is critical.
> Sometimes additional manual steps are required.

Reviewer: ```suggestion
Some additional manual steps might be required, and this information is only available in the release notes.
For important extra information about your upgrade, see the release notes at https://www.uyuni-project.org/pages/stable-version.html.
```

====

Run the [command]``/usr/lib/susemanager/bin/server-migrator.sh`` script to upgrade the base OS and {productname} Server.

[NOTE]
====

> Before running the [command]``server-migrator.sh`` script, check whether the most recent version of the [package]``susemanager`` package is installed:

Reviewer: I would take this info out of the admon, put it above the 'run the script` instruction, and turn it into a proper procedure.

Author: ok

+
----
zypper ref
zypper up susemanager
----
====

> The [command]``server-migrator.sh`` script will migrate {productname} Server to the latest version.

Reviewer: ```suggestion
The [command]``server-migrator.sh`` script migrates {productname} Server to the latest version.
```

> This implies upgrading the underlying operating system to version 15.2.

Reviewer: ```suggestion
It also upgrades the underlying operating system to version 15.2.
```
Can we use a version entity here?

[WARNING]
====

> During migration the spacewalk services need to be stopped.
> The script will do this.

Reviewer: ```suggestion
During migration the script will stop the spacewalk services.
```

> After successful migration the {productname} Server needs to be rebooted manually.

Reviewer: ```suggestion
After the migration is complete, you will need to manually reboot the {productname} Server.
```

====

[IMPORTANT]
====

> Because you cannot fix any issues during migration, make sure you have a backup before starting the migration.

Reviewer: ```suggestion
You will not be able to fix issues that arise during the migration.
Ensure you have created a backup before you start the migration.
```

> If you are running {productname} Server on a virtual machine, it is recommended to create a snapshot before performing the migration.

Reviewer: ```suggestion
If you are running {productname} Server on a virtual machine, we recommend that you create a snapshot before you start.
```

> ====

Reviewer: is there any way we can put all these admons above the migration procedure, and maybe take them out of admons except for the most important ones?

.Procedure: Migrating to {productname} 4.1

> The script runs as [systemitem]``root`` and performs the following actions.

Reviewer: ```suggestion
The script runs as [systemitem]``root`` and performs these actions:
```

[server-major-upgrade]
= Server - Major Upgrade

> From time to time, {productname} Server upgrades are labeled as major.

Author: I'm wondering whether it would be handy to assign some kind of such a label to the Uyuni releases. For the time being, I'll rephrase this (done). I'd like to keep the product documentation version agnostic as much as possible.  I think it's sufficient if you'll mention the version numbers in the release notes.

> From time to time, {productname} Server upgrades are labeled as major.

Reviewer 2: I agree, no need to mention the versions in the doc. I was just giving an example to show that, in the end, we don't have such distinction between minor an major upgrades.

This is especially the case, if a version upgrade of openSUSE Leap is needed. openSUSE Leap is the underlying base operating system (OS).
For background information about upgrading openSUSE Leap, see https://doc.opensuse.org/documentation/leap/startup/single-html/book-opensuse-startup/index.html#sec-update-zypper.

> 

Author: I generalized the ref.  Or should I remove it completely?

> 

Reviewer 2: I'd remove it. Think about Uyuni Server as an full product, just as SUSE Manager is. You update Uyuni or SUSE manager. You don't update Leap or SLE, despite they are using them as base OS.

> Run the [command]``/usr/lib/susemanager/bin/server-migrator.sh`` script to upgrade the base OS and {productname} Server.

Reviewer 2: Is this what you did during the testing? I am pretty sure you had to update a RPM package (susemanager) first.

Author: a package update was not available.  That's why I copy-pasted the commands on the command line (and replaced stable with master).

Reviewer 2: Well, it wasn't because we didn't release yet, but *you need it* on a real situation :-)

[NOTE]
====

> If the [command]``server-migrator.sh`` script is not already installed on your {productname} Server system, update the [package]``susemanager`` package:

Reviewer 2: The script is always installed since Uyuni 4.0.2. But you *always* need to update `susemanager` before running the script!

Author: ok, change.  Maybe, I'd better recommend to do general package update (maintenance resp. online update first).

Reviewer 2: No, don't do that. During such upgrades you need to update `susemanager` first, then run the script. In that precise order.

That is because the script will change your repositories (to use the new Leap version), will perform an upgrade with vendor change allowed and (not this time) will fix any packages incompatibilities for you.

+
----
zypper ref
zypper up susemanager
----
====

The [command]``server-migrator.sh`` script will migrate {productname} Server to the latest version.
This implies upgrading the underlying operating system to version 15.2.

> // During migration the services need to be shut down and after successful migration the server needs to be rebooted manually.

Reviewer 2: I think this should not be commented, and most probably it should be a warning.

Author: ok, it's rephrased and enabled now.

[IMPORTANT]
====
Because you cannot fix any issues during migration, make sure you have a backup before starting the migration.
If you are running {productname} Server on a virtual machine, it is recommended to create a snapshot before performing the migration.
====

.Procedure:

The script runs as [systemitem]``root`` and performs the following actions.

> . Stop the spacewalk services:

Reviewer 2: I don't think this section is explaining the procedure.

The procedure is what you wrote above.

And why do we need to explain the script line by line? Would be enough if you list what it does, but without adding the lines, FMPOV.

What value adds having the script content?

Author: I thought it would help the user if user wants to write an own version of the script. But if you think it will cause confusion, I can remove it.
BTW, in those scripts we should always use long options for readability.

Reviewer 2: The user does not need to write their own version. They could mess the migration.

What do you mean with long options?

Reviewer 2: Why was this conversation merged, the PR merged, and the script content still there?
